crates/sqlbuiltins/src/functions/mod.rs

```diff
@@ -34,9 +42,57 @@ use table::{BuiltinTableFuncs, TableFunc};
 
 use self::alias_map::AliasMap;
 use crate::functions::scalars::openai::OpenAIEmbed;
+use crate::functions::scalars::sentence_transformers::BertUdf;
+use crate::functions::scalars::similarity::CosineSimilarity;
 
 /// All builtin functions available for all sessions.
-pub static FUNCTION_REGISTRY: Lazy<FunctionRegistry> = Lazy::new(FunctionRegistry::new);
+pub static FUNCTION_REGISTRY: Lazy<Mutex<FunctionRegistry>> =
+    Lazy::new(|| Mutex::new(FunctionRegistry::new()));
```

Comment:
Does datafusion require that the function registry be global and mutable at runtime? Of users can load their own functions into a global registry, then users can leak functionality to other users in a multi-user deployment (serverless/etc.) and that doesn't seem particularly tenable. b

I'm not sure what makes sense implementation wise, but I think either the function registry needs to be tied to a users' session, or it should be immutable. 

This might quickly become moot, but is there anyway that the registry itself can manage it's own concurrency control rather than needing to be wrapped in a lock?

Reply:
> Does datafusion require that the function registry be global and mutable at runtime? Of users can load their own functions into a global registry, then users can leak functionality to other users in a multi-user deployment (serverless/etc.) and that doesn't seem particularly tenable. b


yeah that was my comment about moving the function registry into the SessionCatalog. Having it attached to a session is better long term & prevents any accidental leaks between sessions.

Reply:
got it. 

I think having it attached to the session is blocking releasing those code.